components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,848 @@
+            0
+        }
+    }
+
+    // Fixed date of Hebrew new year
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2294
+    #[allow(dead_code)]
+    pub(crate) fn hebrew_new_year(h_year: i32) -> RataDie {
+        RataDie::new(
+            FIXED_HEBREW_EPOCH.to_i64_date()
+                + Self::hebrew_calendar_elapsed_days(h_year) as i64
+                + Self::hebrew_year_length_correction(h_year) as i64,
+        )
+    }
+
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2315
+    #[allow(dead_code)]
+    fn days_in_hebrew_year(h_year: i32) -> u16 {
+        (Self::hebrew_new_year(1 + h_year) - Self::hebrew_new_year(h_year)) as u16
+    }
+
+    // True if the month Marheshvan is going to be long in given Hebrew year
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2321
+    #[allow(dead_code)]
+    fn long_marheshvan(h_year: i32) -> bool {
+        let coll = [355, 385];
+        coll.contains(&Self::days_in_hebrew_year(h_year))
+    }
+    // True if the month Kislve is going to be short in given Hebrew year
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2326
+    #[allow(dead_code)]
+    fn short_kislev(h_year: i32) -> bool {
+        let coll = [353, 383];
+        coll.contains(&Self::days_in_hebrew_year(h_year))
+    }
+
+    // Last day of month (h_month) in Hebrew year (h_year)
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2230
+    #[allow(dead_code)]
+    fn last_day_of_hebrew_month(h_year: i32, h_month: u8) -> u8 {
+        match h_month {
+            IYYAR | TAMMUZ | ELUL | TEVET | ADARII => 29,
+            ADAR => {
+                if !Self::is_leap_year(h_year) {
+                    29
+                } else {
+                    30
+                }
+            }
+            MARHESHVAN => {
+                if !Self::long_marheshvan(h_year) {
+                    29
+                } else {
+                    30
+                }
+            }
+            KISLEV => {
+                if Self::short_kislev(h_year) {
+                    29
+                } else {
+                    30
+                }
+            }
+            _ => 30,
+        }
+    }
+
+    // "Fixed" is a day count representation of calendars staring from Jan 1st of year 1 of the Georgian Calendar.
+    // The fixed date algorithms are from
+    // Dershowitz, Nachum, and Edward M. Reingold. _Calendrical calculations_. Cambridge University Press, 2008.
+    //
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2331
+    #[allow(dead_code)]
+    fn fixed_from_hebrew(date: HebrewDateInner) -> RataDie {
+        let year = date.0.year;
+        let month = date.0.month;
+        let day = date.0.day;
+
+        // Days so far this month.
```

Comment:
Only the subexpression `day.int() - 1` represents the days so far this month, right? If so, either this comment or the code should be changed so that they match

Reply:
yes that is correct, ill fix it.